src/Italy/Mews.Fiscalizations.Italy/Uniwix/Communication/UniwixClient.cs

```diff
@@ -79,52 +74,56 @@ public Task<UniwixUser> CreateUserAsync(CreateUserParameters createUserParameter
             return PostAsync<UniwixUser>(url, content);
         }
 
-        public Task<bool> VerifyCredentialsAsync()
+        public async Task<ITry<bool, ErrorResult>> VerifyCredentialsAsync()
         {
-            var url = $"{UniwixBaseUrl}/Info";
-            return ExecuteRequestAsync(url, HttpMethod.Get, content: null, responseProcessor: r => r.IsSuccessStatusCode);
+            // This endpoint returns the account information, so if the ITry is success it means the credentials are valid.
+            var result = await GetAsync<object>($"{UniwixBaseUrl}/Info");
+            return result.Map(r => true);
         }
 
-        private Task<TResult> GetAsync<TResult>(string url)
+        private Task<ITry<TResult, ErrorResult>> GetAsync<TResult>(string url)
         {
             return ExecuteRequestAsync<TResult>(url, HttpMethod.Get, content: null);
         }
 
-        private Task<TResult> PostAsync<TResult>(string url, HttpContent content)
+        private Task<ITry<TResult, ErrorResult>> PostAsync<TResult>(string url, HttpContent content)
         {
             return ExecuteRequestAsync<TResult>(url, HttpMethod.Post, content);
         }
 
-        private async Task<TResult> ExecuteRequestAsync<TResult>(string url, HttpMethod httpMethod, HttpContent content)
+        private Task<ITry<TResult, ErrorResult>> ExecuteRequestAsync<TResult>(string url, HttpMethod httpMethod, HttpContent content)
         {
-            var response = await ExecuteRequestAsync(url, httpMethod, content, async httpResponse =>
+            return ExecuteRequestAsync(url, httpMethod, content, httpResponse =>
             {
-                var json = await httpResponse.Content.ReadAsStringAsync();
+                var json = httpResponse.Content.ReadAsStringAsync().Result;
```

Comment:
I would like to double check whether it is intentional. When Result of incomplete task is accessed I believe it would make this operation synchronous. Could you please consider keeping it async and await on line 142?